src/app/service/auth/auth-service.service.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+@Injectable({
+  providedIn: 'root'
+})
+
+export class AuthService {
+
+  constructor(private httpClient: HttpClient) { }
+
+  public async authenticateUser(userName: string, password: string) {
```

Comment:
can you change these to return observables now ? Because observables are better than promises

and each function should have a return type.